flink-ml-lib/src/test/java/org/apache/flink/ml/classification/OnlineModelSaveLoadTest.java

```diff
@@ -0,0 +1,178 @@
+    private String tmpPath;
+
+    @Before
+    public void before() {
+        Configuration config = new Configuration();
+        config.set(ExecutionCheckpointingOptions.ENABLE_CHECKPOINTS_AFTER_TASKS_FINISH, true);
+        env = StreamExecutionEnvironment.getExecutionEnvironment(config);
+        env.enableCheckpointing(100);
+        env.setRestartStrategy(RestartStrategies.noRestart());
+        env.setParallelism(1);
+        tEnv = StreamTableEnvironment.create(env);
+        Schema modelSchema =
+                Schema.newBuilder().column("f0", DataTypes.of(DenseVector.class)).build();
+        Table modelDataTable = tEnv.fromDataStream(env.fromCollection(modelRows), modelSchema);
+        try {
+            tmpPath = tempFolder.newFolder().getAbsolutePath();
+            /* Saves online model data to given path (tmpPath). */
+            saveModelData(
+                    LogisticRegressionModelData.getModelDataStream(modelDataTable),
+                    tmpPath,
+                    new LogisticRegressionModelData.ModelDataEncoder());
+            env.execute();
+        } catch (Exception e) {
+            throw new RuntimeException("Saves model data failed!");
+        }
+    }
+
+    @Test
+    public void testSaveAndLoadSingleModel() throws Exception {
+        Process proc = Runtime.getRuntime().exec("ls " + tmpPath + "/data");
+        BufferedReader bufferedReader =
+                new BufferedReader(new InputStreamReader(proc.getInputStream()));
+
+        /* Loads every LogisticRegression model in model path. */
+        String modelVersion;
+        while ((modelVersion = bufferedReader.readLine()) != null) {
+            Table modelData =
+                    tEnv.fromDataStream(
+                            loadModelData(
+                                    env,
+                                    tmpPath,
+                                    new LogisticRegressionModelData.ModelDataDecoder(),
+                                    modelVersion));
+
+            List<Row> loadedModelRows =
+                    IteratorUtils.toList(tEnv.toDataStream(modelData).executeAndCollect());
+            int numEquals = 0;
+            for (int i = 0; i < modelRows.size(); ++i) {
+                if (modelRows.get(i).equals(loadedModelRows.get(0))) {
+                    numEquals++;
+                }
+            }
+            Assert.assertEquals(1, numEquals);
+        }
+    }
+
+    @Test
+    public void testSaveAndLoadOnlineModel() throws Exception {
+        Table modelData =
+                tEnv.fromDataStream(
+                        ReadWriteUtils.loadModelData(
+                                env, tmpPath, new LogisticRegressionModelData.ModelDataDecoder()));
+
+        List<Row> loadedModelRows =
+                IteratorUtils.toList(tEnv.toDataStream(modelData).executeAndCollect());
+        int numEquals = 0;
+        for (int i = 0; i < loadedModelRows.size(); ++i) {
```

Comment:
Can you refine the check using `TestBaseUtils#compareResultCollections` here?

The test may still success even if the loaded model data is wrong. (e.g., four model data with version 0)

Reply:
OK